File: 000_C# Training/C#_2 Classes and methods/VB.NET/classesandmethods.vb
Option Strict

Public Module HomeModule
    Public Sub Main()
        MethodInTheSameModule
        OtherModule.MethodInAnotherModule
        
        Dim returnValue As String
        returnValue = MethodReturningValue()
        Console.WriteLine(returnValue)
        
        MethodReceivingValue("XYZ")

        Console.ReadLine()
    End Sub

    Public Sub MethodInTheSameModule()
        Console.WriteLine("Hello, i am the method in the same module. I don't return anything, that's why I am called 'Sub' in VBA. My C# counterpart would be 'static void'")
    End Sub

    Public Function MethodReturningValue() As String
        Console.WriteLine("Hello, I return a string, that's why I am called 'Function' in VBA. My C# counterpart would be 'static string'")

        MethodReturningValue = "ABC"
    End Function

    Public Sub MethodReceivingValue(receivedString As String)
        Console.WriteLine("Hello, I received the string " + receivedString + ".")
    End Sub
End Module

Public Module OtherModule
    Public Sub MethodInAnotherModule()
        Console.WriteLine("Hello, I am the method in another module")
    End Sub
End Module

File: 000_C# Training/C#_2 Classes and methods/C#/classesandmethods.cs
using System;

class MainClass
{
    public static void Main()
    {
        MethodInTheSameClass();
        OtherClass.MethodInAnotherClass();
        string returnValue = MethodReturningValue();
        Console.WriteLine(returnValue);
        
        MethodReceivingValue("XYZ");

        Console.ReadLine();
    }

    public static void MethodInTheSameClass()
    {
        Console.WriteLine("Hello, i am the method in the same class. I am void meaning that I don't return anything. My VBA counterpart would be 'Sub'");
    }

    public static string MethodReturningValue()
    {
        Console.WriteLine("Hello, I return a string. So I need at least one time the key word 'return' followed by the string that I return. My VBA counterpart would be 'Function'");
        return "ABC";
    }

    public static void MethodReceivingValue(string receivedString)
    {
        Console.WriteLine("Hello, I received the string " + receivedString + ".");
    }   
}

class OtherClass
{
    public static void MethodInAnotherClass()
    {
        Console.WriteLine("Hello, I am the method in another class");
    }
}